src/as_execution/mod.rs

```diff
@@ -129,7 +133,9 @@ pub(crate) fn init_sp_engine(limit: u64, gas_costs: GasCosts, max_exports: usize
     compiler_config.canonicalize_nans(true);
 
     // Add condom middleware
-    compiler_config.push_middleware(Arc::new(CondomMiddleware::new(max_exports)));
+    if let Some(max_exports) = max_exports {
```

Comment:
Don't remove the condom, leave it on.
In the futrue there might be extra checks performed inside.
Instead, pass an Option to the condom, and ignore this particular check inside of it if it is None.

Same for WASMV1 and for AS

Reply:
Ok, seems good!